File: leetcode19/leetcode19.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dumpHead = ListNode()
        dumpHead.next = head
        cnt = 0
        tail = dumpHead
        cur = dumpHead
        while tail.next != None:
            tail = tail.next
            cnt += 1
        cnt -= n
        for _ in range(cnt):
            cur = cur.next
        cur.next = cur.next.next
        return dumpHead.next

File: leetcode19/leetcode19.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

#include<bits/stdc++.h>
using namespace std;
struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* dummy = new ListNode();
        dummy->next = head;
        int cnt = 0;
        ListNode* cur = dummy;
        ListNode* pre = dummy;
        while (pre->next != nullptr)
        {
            pre = pre->next;
            cnt++;
        }
        cnt -= n;
        while (cnt--)
        {
            cur = cur->next;
        }
        ListNode* tmp = cur->next;
        cur->next = cur->next->next;
        delete tmp;
        return dummy->next;
        
        
    }
};